### DISCUSSION TITLE
Which files are not needed, if using Laravel for API development.

### DISCUSSION THREAD

**Discussion by tonyjemba** [laravel/framework#39484] [Q&A]
I was thinking of minifying my project, I want to know which files are not need if using Laravel for API development.
I am using Laravel sail.

**Comment by vuvanly** [comment]
If you are going to develop API, I suggest that you can use [Lumen framework](https://lumen.laravel.com/). It's framework based on Laravel and was optimized for API development. We're using Lumen right now.

**Comment by dennisprudlo** [comment]
I would still use Laravel instead of Lumen. Even Taylor said it makes more sense to use Laravel to get all the features we love. In some special cases where a faster boot is required and the missing features are not a problem, Lumen may be the right fit. https://twitter.com/taylorotwell/status/1441101127496323080?lang=en

But as it's said in the tweet:
> be sure that you *actually* need this [speed boost] first!

In most cases the Laravel application boots fast enough, especially if it's a regular api request from a user.

**Reply by vuvanly** [reply]
@dennisprudlo just curious but if laravel/octane support Lumen as well, so Lumen will become superfast?

**Reply by dennisprudlo** [reply]
As far as I understand the reason why lumen is "so fast" is because it has much less to boot whenever a request comes in. But Octane boots the application once and stores it in memory. So using Octane + Laravel or Octane + Lumen should should be the same in terms of speed. Everything after framework boot is your domain. Eloquent will still need the same amount of time to hydrate models in lumen as it does in laravel.

**Reply by vuvanly** [reply]
@dennisprudlo I got it now. The different between Laravel and Lumen is booting time, but now Octane load it and store it in memory, so both Laravel and Lumen will be able to skip the booting time. Thank you for your explanation.
